modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java

```diff
@@ -298,6 +330,51 @@ public void testGetSegmentsEntryIdsWithContextCriterionAndModelCriterion()
 				segmentsEntryIds));
 	}
 
+	@Test
+	public void testGetSegmentsEntryIdsWithContextCriterionAndModelCriterionAndDefaultUser()
+		throws Exception {
+
+		Company company = _companyLocalService.getCompany(
+			TestPropsValues.getCompanyId());
+
+		User defaultUser = company.getDefaultUser();
+
+		Criteria criteria = new Criteria();
+
+		_contextSegmentsCriteriaContributor.contribute(
+			criteria, "(languageId eq 'en')", Criteria.Conjunction.AND);
+
+		_userSegmentsCriteriaContributor.contribute(
+			criteria,
+			String.format("(firstName eq '%s')", defaultUser.getFirstName()),
+			Criteria.Conjunction.AND);
+
+		SegmentsTestUtil.addSegmentsEntry(
+			_group.getGroupId(), CriteriaSerializer.serialize(criteria),
+			User.class.getName());
+
+		Context context = new Context();
+
+		context.put(Context.LANGUAGE_ID, "en");
+		context.put(Context.SIGNED_IN, false);
+
+		long[] segmentsEntryIds = _segmentsEntryProvider.getSegmentsEntryIds(
+			_group.getGroupId(), User.class.getName(), defaultUser.getUserId(),
+			context);
+
+		Assert.assertArrayEquals(new long[0], segmentsEntryIds);
+
+		context = new Context();
+
+		context.put(Context.LANGUAGE_ID, "en");
+
+		segmentsEntryIds = _segmentsEntryProvider.getSegmentsEntryIds(
```

Comment:
As I said before, you should not call the method twice in the same test, please could you create a new test for this specific use case. Let me know if I can help

Reply:
Ok, but we are going to have all the DefaultUser tests duplicated with and without the `context.put(Context.SIGNED_IN, false);` parameter